### PR TITLE
* Updated docs for v12.

### PR DESCRIPTION
**Pull request by dpetzold**
* Added tolist call to fix error:

```
Error: Error in function call

  on modules/info/main.tf line 27, in data "aws_subnet" "destination":
  27:   id    = element(data.aws_subnet_ids.destination.ids, count.index)
    |----------------
    | data.aws_subnet_ids.destination.ids is set of string with 2 elements
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Helps #7522

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by bbaptist** [comment]
This change should also address that the docs say it returns a list.

"aws_subnet_ids provides a list of ids for a vpc_id"

**Comment by Dzhuneyt** [comment]
@dpetzold it seems there are some conflicts. Could you please take a look? Thank you for your contribution!

**Comment by dpetzold** [comment]
@Dzhuneyt the PR has been updated. Below is the output from the code sample:

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

subnet_cidr_blocks = [
  "172.18.12.128/25",
  "172.18.14.0/25",
  "172.18.13.0/25",
  "172.18.14.128/25",
  "172.18.13.128/25",
  "172.18.12.0/25",
]
```

Thanks for looking at this!

File: website/docs/d/subnet_ids.html.markdown [hunk 1]
```diff
@@ -10,24 +10,24 @@ description: |-
...
+This resource can be useful for getting back a set of subnet ids for a vpc.
 
 ## Example Usage
 
 The following shows outputing all cidr blocks for every subnet id in a vpc.
 
 ```hcl
-data "aws_subnet_ids" "example" {
-  vpc_id = "${var.vpc_id}"
+data aws_subnet_ids example {
```
Comment (gdavison) [inline website/docs/d/subnet_ids.html.markdown]: Please leave the quotes on the `data` and `output` blocks here and in the last code section.

File: website/docs/d/subnet_ids.html.markdown [hunk 2]
```diff
@@ -10,24 +10,24 @@ description: |-
...
+  vpc_id = var.vpc_id
 }
 
-data "aws_subnet" "example" {
-  count = "${length(data.aws_subnet_ids.example.ids)}"
-  id    = "${data.aws_subnet_ids.example.ids[count.index]}"
+data aws_subnet example {
+  count = length(data.aws_subnet_ids.example.ids)
```
Comment (gdavison) [inline website/docs/d/subnet_ids.html.markdown]: Please use `for_each` here instead of `count`

File: website/docs/d/subnet_ids.html.markdown [hunk 3]
```diff
@@ -10,24 +10,24 @@ description: |-
...
+  id    = tolist(data.aws_subnet_ids.example.ids)[count.index]
 }
 
-output "subnet_cidr_blocks" {
-  value = "${data.aws_subnet.example.*.cidr_block}"
+output subnet_cidr_blocks {
+  value = data.aws_subnet.example.*.cidr_block
```
Comment (gdavison) [inline website/docs/d/subnet_ids.html.markdown]: This should be replaced with `[for s in data.aws_subnet.for-each-example : s.cidr_block]` when updated to use `for_each`

File: website/docs/d/subnet_ids.html.markdown [hunk 4]
```diff
@@ -46,9 +46,9 @@ data "aws_subnet_ids" "private" {
 
 resource "aws_instance" "app" {
   count         = "3"
```
Comment (gdavison) [inline website/docs/d/subnet_ids.html.markdown]: Use `for_each` here as well.

**Pull request merged**